src/zcash/circuit/utils.tcc

```diff
@@ -47,9 +46,26 @@ void insert_uint64(std::vector<bool>& into, uint64_t from) {
     into.insert(into.end(), num.begin(), num.end());
 }
 
+template<typename T>
+T swap_endianness_u64(T v) {
+    if (v.size() != 64) {
+        throw std::length_error("invalid bit length for 64-bit unsigned integer");
+    }
+
+    for (size_t i = 0; i < 4; i++) {
+        for (size_t j = 0; j < 8; j++) {
+            std::swap(v[i*8 + j], v[((7-i)*8)+j]);
+        }
+    }
+
+    return v;
+}
+
 template<typename FieldT>
-linear_combination<FieldT> packed_addition(pb_variable_array<FieldT>& input) {
+linear_combination<FieldT> packed_addition(pb_variable_array<FieldT> input) {
+    auto input_swapped = swap_endianness_u64(input);
+
     return pb_packing_sum<FieldT>(pb_variable_array<FieldT>(
-        input.rbegin(), input.rend()
+        input_swapped.rbegin(), input_swapped.rend()
```

Comment:
Hmm, this is ugly. The input is represented as mixed-endian. When I accepted the switch from being consistently little-endian, to using little-endian byte order but with the MS bit first in a byte, I hadn't anticipated that we'd need to do that. I had thought that we'd keep all the value fields in some consistent order (doesn't matter which), and do the swapping only for the representation of v in the commitment input.